src/SBVonKarman.cpp

```diff
@@ -0,0 +1,395 @@
+    //
+    //
+    //
+
+    const double VonKarmanInfo::magic1 = 2*boost::math::tgamma(11./6)/(pow(2, 5./6)*pow(M_PI, 8./3))
+                                    * pow(24/5.*boost::math::tgamma(6./5), 5./6);
+    const double VonKarmanInfo::magic2 = boost::math::tgamma(5./6)/pow(2., 1./6);
+    const double VonKarmanInfo::magic3 = VonKarmanInfo::magic1*boost::math::tgamma(-5./6)/pow(2., 11./6);
+    const double VonKarmanInfo::magic4 = boost::math::tgamma(11./6)*boost::math::tgamma(5./6)
+                                    / pow(M_PI,8./3)
+                                    * pow(24./5*boost::math::tgamma(6./5),5./6);
+
+    class VKIkValueResid {
+    public:
+        VKIkValueResid(const VonKarmanInfo& vki, double mkt) : _vki(vki), _mkt(mkt) {}
+        double operator()(double k) const {
+            double val = _vki.kValue(k)-_mkt;
+            xdbg<<"resid(k="<<k<<")="<<val<<'\n';
+            return val;
+         }
+    private:
+        const double _mkt;
+        const VonKarmanInfo& _vki;
+    };
+
+    VonKarmanInfo::VonKarmanInfo(double lam, double r0, double L0, bool doDelta,
+                                 const GSParamsPtr& gsparams) :
+        _lam(lam), _r0(r0), _L0(L0), _r0L0m53(pow(r0/L0, -5./3)), _gsparams(gsparams),
+        _deltaAmplitude(exp(-0.5*magic4*_r0L0m53)),
+        _doDelta(doDelta),
+        _radial(TableDD::spline)
+    {
+        // determine maxK
+        // want kValue(maxK)/kValue(0.0) = _gsparams->maxk_threshold;
+        // note that kValue(0.0) = 1.
+        double mkt = _gsparams->maxk_threshold;
+        if (_doDelta) {
+            if (mkt < _deltaAmplitude) {
+                // If the delta function amplitude is too large, then no matter how far out in k we
+                // go, kValue never drops below that amplitude.
+                // _maxk = std::numeric_limits<double>::infinity();
+                _maxk = MOCK_INF;
+            } else {
+                mkt = mkt*(1-_deltaAmplitude)+_deltaAmplitude;
+            }
+        }
+        // if (_maxk != std::numeric_limits<double>::infinity()) {
+        if (_maxk != MOCK_INF) {
+            VKIkValueResid vkikvr(*this, mkt);
+            Solve<VKIkValueResid> solver(vkikvr, 0.1, 1);
+            solver.bracket();
+            solver.setMethod(Brent);
+            _maxk = solver.root();
+        }
+        dbg<<"_maxk = "<<_maxk<<" arcsec^-1\n";
+        dbg<<"SB(maxk) = "<<kValue(_maxk)<<'\n';
+        dbg<<"_deltaAmplitude = "<<_deltaAmplitude<<'\n';
+
+        // build the radial function, and along the way, set _stepk, _hlr.
+        _buildRadialFunc();
+    }
+
+    double VonKarmanInfo::structureFunction(double rho) const {
+    // rho in meters
+        double rhoL0 = rho/_L0;
+        if (rhoL0 < 1e-6) {
+            return -magic3*fast_pow(2*M_PI*rho/_r0, 5./3);
+        } else {
+            double x = 2*M_PI*rhoL0;
+            return magic1*_r0L0m53*(magic2-fast_pow(x, 5./6)*boost::math::cyl_bessel_k(5./6, x));
```

Comment:
Similarly, this is the only place magic1,2,3 are used.  These could be static const variables inside this function (which the compiler will compute at compile time, so no runtime cost), or global const variables defined just before this function.  Either way, it would be helpful to future developers looking at this code to see the definitions on the same screen as where they are used.